omod/src/test/java/org/openmrs/module/idgen/rest/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,206 @@
+		autoGenerationOption.add("identifierType", IDENTIFIER_TYPE_UUID);
+		autoGenerationOption.add("location", LOCATION);
+		autoGenerationOption.add("source", SOURCE);
+		autoGenerationOption.add("automaticGenerationEnabled", true);
+		autoGenerationOption.add("manualEntryEnabled", true);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		SimpleObject response = deserialize(handle(req));
+		assertNotNull(PropertyUtils.getProperty(response, "uuid"));
+		assertEquals(originalCount + 1, getAllCount());
+		Object autogenerationOptionUuid = PropertyUtils.getProperty(response, "uuid");
+		AutoGenerationOption newAutoGenerationOption = identifierSourceService
+		        .getAutoGenerationOption(Integer.parseInt(autogenerationOptionUuid.toString()));
+		assertEquals(SOURCE, newAutoGenerationOption.getSource().getUuid());
+		assertEquals(LOCATION, newAutoGenerationOption.getLocation().getUuid());
+		assertEquals(IDENTIFIER_TYPE_UUID, newAutoGenerationOption.getIdentifierType().getUuid());
+		assertEquals(true, newAutoGenerationOption.isAutomaticGenerationEnabled());
+		assertEquals(true, newAutoGenerationOption.isManualEntryEnabled());
+	}
+	
+	@Test()
+	public void shouldCreateNewAutoGenerationOptionWithoutLocation() throws Exception {
+		long originalCount = getAllCount();
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", IDENTIFIER_TYPE_UUID);
+		autoGenerationOption.add("source", SOURCE);
+		autoGenerationOption.add("automaticGenerationEnabled", true);
+		autoGenerationOption.add("manualEntryEnabled", true);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		SimpleObject response = deserialize(handle(req));
+		assertNotNull(PropertyUtils.getProperty(response, "uuid"));
+		assertEquals(originalCount + 1, getAllCount());
+		Object autogenerationOptionUuid = PropertyUtils.getProperty(response, "uuid");
+		AutoGenerationOption newAutoGenerationOption = identifierSourceService
+		        .getAutoGenerationOption(Integer.parseInt(autogenerationOptionUuid.toString()));
+		assertEquals(SOURCE, newAutoGenerationOption.getSource().getUuid());
+		assertNull(newAutoGenerationOption.getLocation());
+		assertEquals(IDENTIFIER_TYPE_UUID, newAutoGenerationOption.getIdentifierType().getUuid());
+		assertEquals(true, newAutoGenerationOption.isAutomaticGenerationEnabled());
+		assertEquals(true, newAutoGenerationOption.isManualEntryEnabled());
+	}
+	
+	@Test(expected = ValidationException.class)
+	public void shouldNotCreateNewAutoGenerationOptionWithNullParameters() throws Exception {
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", null);
+		autoGenerationOption.add("location", null);
```

Comment:
For null values, we just do not supply those properties in the autoGenerationOption